pkg/kubelet/kubeletconfig/configsync.go

```diff
@@ -62,13 +77,22 @@ func (cc *Controller) syncConfigSource(client clientset.Interface, nodeName stri
 	}
 
 	// check the Node and download any new config
-	if updated, reason, err := cc.doSyncConfigSource(client, node.Spec.ConfigSource); err != nil {
+	if updated, cur, reason, err := cc.doSyncConfigSource(client, node.Spec.ConfigSource); err != nil {
 		cc.configOK.SetFailSyncCondition(reason)
 		syncerr = fmt.Errorf("%s, error: %v", reason, err)
 		return
 	} else if updated {
-		// TODO(mtaufen): Consider adding a "currently restarting kubelet" ConfigOK message for this case
-		utillog.Infof("config updated, Kubelet will restart to begin using new config")
+		path := LocalConfigMessage
+		if cur != nil {
+			path = cur.APIPath()
+		}
+		// we directly log and send the event, instead of using the event recorder,
+		// because the event recorder won't flush its queue before we exit (we'd lose the event)
```

Comment:
Should kubelet shut down more gracefully?

Reply:
We had a long discussion about this during the dynamic config design phase, and decided it's too complex to do a graceful shutdown in today's Kubelet. It's in our list of things to maybe do in the future.

Reply:
Ack